edk2toollib/uefi/edk2/build_objects/dsc_translator_test.py

```diff
@@ -0,0 +1,500 @@
+  MdeModulePkg/Universal/Network/Ip4ConfigDxe/Ip4ConfigDxe.inf
+  MdeModulePkg/Universal/Network/Ip4Dxe/Ip4Dxe.inf
+  MdeModulePkg/Universal/Network/MnpDxe/MnpDxe.inf
+  MdeModulePkg/Universal/Network/VlanConfigDxe/VlanConfigDxe.inf
+  MdeModulePkg/Universal/Network/Mtftp4Dxe/Mtftp4Dxe.inf
+  MdeModulePkg/Universal/Network/Tcp4Dxe/Tcp4Dxe.inf
+  MdeModulePkg/Universal/Network/Udp4Dxe/Udp4Dxe.inf
+  MdeModulePkg/Universal/Network/UefiPxeBcDxe/UefiPxeBcDxe.inf
+  Nt32Pkg/SnpNt32Dxe/SnpNt32Dxe.inf
+  MdeModulePkg/Universal/Network/IScsiDxe/IScsiDxe.inf
+  IntelFrameworkModulePkg/Universal/BdsDxe/BdsDxe.inf
+  MdeModulePkg/Universal/HiiDatabaseDxe/HiiDatabaseDxe.inf
+  MdeModulePkg/Universal/SetupBrowserDxe/SetupBrowserDxe.inf
+  MdeModulePkg/Universal/PrintDxe/PrintDxe.inf
+  MdeModulePkg/Universal/DriverSampleDxe/DriverSampleDxe.inf {
+    <LibraryClasses>
+      PcdLib|MdePkg/Library/BasePcdLibNull/BasePcdLibNull.inf
+    <PcdsFeatureFlag>
+      gEfiMdeModulePkgTokenSpaceGuid.PcdStatusCodeUseSerial|TRUE
+  }
+  MdeModulePkg/Application/VariableInfo/VariableInfo.inf
+  MdeModulePkg/Universal/PlatformDriOverrideDxe/PlatformDriOverrideDxe.inf
+
+######################################################################## #
+# BuildOptions Section - Define the module specific tool chain flags that
+# should be used as the default flags for a
+# module. These flags are appended to any
+# standard flags that are defined by the build
+# process. They can be applied for any modules or
+# only those modules with the specific module
+# style (EDK or EDKII) specified in [Components] # section.
+#
+########################################################################
+[BuildOptions]
+  DEBUG_*_IA32_DLINK_FLAGS = /BASE:0x10000 /ALIGN:4096 /FILEALIGN:4096 \\
+                             /EXPORT:InitializeDriver=$(IMAGE_ENTRY_POINT) \\
+                             /SUBSYSTEM:CONSOLE
+  RELEASE_*_IA32_DLINK_FLAGS = /ALIGN:4096 /FILEALIGN:4096
+  *_*_IA32_CC_FLAGS = /D EFI_SPECIFICATION_VERSION = 0x0002000A \\
+                      /D TIANO_RELEASE_VERSION=0x00080006
+
+"""
+
+    def write_file(self, file_path, contents):
+        temp_dir = tempfile.mkdtemp()
+        file_path = os.path.join(temp_dir, file_path)
+        f = open(file_path, "w")
+        f.write(contents)
+        f.close()
+        return file_path
+
+    def test_dsc_to_file_and_back_again(self):
+        filepath = self.write_file("test.dsc", self.test_dsc)
+        print(filepath)
+        # parse the original DSC
+        parser = DscParser()
+        parser._ErrorLimit = 0
+        # TODO: actually parse the file
+        dsc_obj = []  # parser.ParseFile(filepath)
+        # Write out to disk
+        test_path = os.path.join(os.path.dirname(filepath), "test2.dsc")
+        DscTranslator.dsc_to_file(dsc_obj, test_path)
+        # parse in the outputted DSC
+        parser2 = DscParser()
+        parser2._ErrorLimit = 0
+        print(test_path)
+        # TODO: actually parse the file
```

Comment:
Seems like a pretty important TODO... ;)

Reply:
I have a better parser in #50 but I decided to wait until the real dsc parser is merged in